pkg/api/server/db/model.go

```diff
@@ -28,3 +32,32 @@ type Record struct {
 	Name       string `gorm:"index:records_by_name,priority:3"`
 	Data       []byte
 }
+
+// Annotations is a custom-defined type of a gorm model field.
+type Annotations map[string]string
+
+// Scan implements sql.Scanner interface, to let gorm know how to resolve the corresponding data from database.
+func (ann *Annotations) Scan(value interface{}) error {
+	if ann == nil {
+		return errors.New("nil pointer")
+	}
+	bytes, ok := value.([]byte)
+	if !ok {
+		return errors.New(fmt.Sprint("failed to obtain a byte array form database:", value))
```

Comment:
It might be useful to also include the type we don't know what this might be if this fails.

```suggestion
		return errors.New(fmt.Sprintf("wanted []byte, got %T: %+v", value, value))
```

Reply:
Done